### DISCUSSION TITLE
It seems that my model is unable to recognize ocean current flows.

### DISCUSSION THREAD

**Discussion by vannameiKim** [OpenDrift/opendrift#1190] [General]
Hello, my name is Kim, and I currently live in South Korea. I've encountered a problem while working with OpenDrift and am seeking assistance. I am using the HYCOM model to perform particle tracking simulations in the coastal regions of Korea. Despite following the methods presented in the gallery and conducting experiments accordingly, it seems the model is not recognizing ocean currents.

To help understand this issue, I have attached my code and related materials.

I would greatly appreciate any advice on how to resolve this issue.

from opendrift.models.oceandrift import OceanDrift
from datetime import datetime, timedelta

o = OceanDrift(loglevel=20)

# Adding readers for global Thredds datasets:
# - Ocean forecast from global Hycom
# - Weather forecast from NOAA/NCEP
o.add_readers_from_list([
    'http://tds.hycom.org/thredds/dodsC/GLBy0.08/expt_93.0/uv3z',
    'https://pae-paha.pacioos.hawaii.edu/thredds/dodsC/ncep_global/NCEP_Global_Atmospheric_Model_best.ncd'])
21:52:21 INFO    opendrift.models.basemodel:528: OpenDriftSimulation initialised (version 1.10.7 / v1.10.6-166-g42c6938)

o.set_config('general:use_auto_landmask', True)
o.set_config('drift:horizontal_diffusivity', 1.0) 
start_time = datetime(2023, 7, 1, 0, 0)
start_lon, start_lat = 124, 34

o.seed_elements(start_lon, start_lat, radius=0, number=10000, time=start_time)
o.run(duration=timedelta(hours=12), time_step=1800)
21:52:34 INFO    opendrift.models.basemodel.environment:219: Adding a dynamical landmask with max. priority based on assumed maximum speed of 1.0 m/s. Adding a customised landmask may be faster...
21:52:34 INFO    opendrift.models.basemodel.environment:246: Fallback values will be used for the following variables which have no readers: 
21:52:34 INFO    opendrift.models.basemodel.environment:249: 	x_sea_water_velocity: 0.000000
21:52:34 INFO    opendrift.models.basemodel.environment:249: 	y_sea_water_velocity: 0.000000
21:52:34 INFO    opendrift.models.basemodel.environment:249: 	x_wind: 0.000000
21:52:34 INFO    opendrift.models.basemodel.environment:249: 	y_wind: 0.000000
21:52:34 INFO    opendrift.models.basemodel.environment:249: 	upward_sea_water_velocity: 0.000000
21:52:34 INFO    opendrift.models.basemodel.environment:249: 	ocean_vertical_diffusivity: 0.000000
21:52:34 INFO    opendrift.models.basemodel.environment:249: 	sea_surface_wave_significant_height: 0.000000
21:52:34 INFO    opendrift.models.basemodel.environment:249: 	sea_surface_wave_stokes_drift_x_velocity: 0.000000
21:52:34 INFO    opendrift.models.basemodel.environment:249: 	sea_surface_wave_stokes_drift_y_velocity: 0.000000
21:52:34 INFO    opendrift.models.basemodel.environment:249: 	sea_surface_wave_period_at_variance_spectral_density_maximum: 0.000000
21:52:34 INFO    opendrift.models.basemodel.environment:249: 	sea_surface_wave_mean_period_from_variance_spectral_density_second_frequency_moment: 0.000000
21:52:34 INFO    opendrift.models.basemodel.environment:249: 	sea_surface_swell_wave_to_direction: 0.000000
21:52:34 INFO    opendrift.models.basemodel.environment:249: 	sea_surface_swell_wave_peak_period_from_variance_spectral_density: 0.000000
21:52:34 INFO    opendrift.models.basemodel.environment:249: 	sea_surface_swell_wave_significant_height: 0.000000
21:52:34 INFO    opendrift.models.basemodel.environment:249: 	sea_surface_wind_wave_to_direction: 0.000000
21:52:34 INFO    opendrift.models.basemodel.environment:249: 	sea_surface_wind_wave_mean_period: 0.000000
21:52:34 INFO    opendrift.models.basemodel.environment:249: 	sea_surface_wind_wave_significant_height: 0.000000
21:52:34 INFO    opendrift.models.basemodel.environment:249: 	surface_downward_x_stress: 0.000000
21:52:34 INFO    opendrift.models.basemodel.environment:249: 	surface_downward_y_stress: 0.000000
21:52:34 INFO    opendrift.models.basemodel.environment:249: 	turbulent_kinetic_energy: 0.000000
21:52:34 INFO    opendrift.models.basemodel.environment:249: 	turbulent_generic_length_scale: 0.000000
21:52:34 INFO    opendrift.models.basemodel.environment:249: 	ocean_mixed_layer_thickness: 50.000000
21:52:34 INFO    opendrift.models.basemodel.environment:249: 	sea_floor_depth_below_sea_level: 10000.000000
21:52:34 INFO    opendrift.models.basemodel:891: Using existing reader for land_binary_mask
21:52:34 INFO    opendrift.models.basemodel:903: All points are in ocean
21:52:34 INFO    opendrift.models.basemodel:1989: 2023-07-01 00:00:00 - step 1 of 24 - 10000 active elements (0 deactivated)
21:52:35 INFO    opendrift.readers.reader_netCDF_CF_generic:92: Opening dataset: http://tds.hycom.org/thredds/dodsC/GLBy0.08/expt_93.0/uv3z
21:53:06 INFO    opendrift.readers.reader_netCDF_CF_generic:253: Grid coordinates are detected, but proj4 string not given: assuming latlong
21:53:06 INFO    opendrift.readers.basereader:166: Variable x_sea_water_velocity will be rotated from eastward_sea_water_velocity
21:53:06 INFO    opendrift.readers.basereader:166: Variable y_sea_water_velocity will be rotated from northward_sea_water_velocity
21:53:07 INFO    opendrift.readers.reader_netCDF_CF_generic:92: Opening dataset: https://pae-paha.pacioos.hawaii.edu/thredds/dodsC/ncep_global/NCEP_Global_Atmospheric_Model_best.ncd
21:53:13 INFO    opendrift.readers.basereader:166: Variable x_wind will be rotated from eastward_wind
21:53:13 INFO    opendrift.readers.basereader:166: Variable y_wind will be rotated from northward_wind
21:53:42 INFO    opendrift.models.basemodel:1989: 2023-07-01 00:30:00 - step 2 of 24 - 10000 active elements (0 deactivated)
21:54:04 INFO    opendrift.models.basemodel:1989: 2023-07-01 01:00:00 - step 3 of 24 - 10000 active elements (0 deactivated)
21:54:04 INFO    opendrift.models.basemodel:1989: 2023-07-01 01:30:00 - step 4 of 24 - 10000 active elements (0 deactivated)
21:54:04 INFO    opendrift.models.basemodel:1989: 2023-07-01 02:00:00 - step 5 of 24 - 10000 active elements (0 deactivated)
21:54:04 INFO    opendrift.models.basemodel:1989: 2023-07-01 02:30:00 - step 6 of 24 - 10000 active elements (0 deactivated)
21:54:04 INFO    opendrift.models.basemodel:1989: 2023-07-01 03:00:00 - step 7 of 24 - 10000 active elements (0 deactivated)
21:54:04 INFO    opendrift.models.basemodel:1989: 2023-07-01 03:30:00 - step 8 of 24 - 10000 active elements (0 deactivated)
21:56:50 INFO    opendrift.models.basemodel:1989: 2023-07-01 04:00:00 - step 9 of 24 - 10000 active elements (0 deactivated)
21:56:50 INFO    opendrift.models.basemodel:1989: 2023-07-01 04:30:00 - step 10 of 24 - 10000 active elements (0 deactivated)
21:56:50 INFO    opendrift.models.basemodel:1989: 2023-07-01 05:00:00 - step 11 of 24 - 10000 active elements (0 deactivated)
21:56:50 INFO    opendrift.models.basemodel:1989: 2023-07-01 05:30:00 - step 12 of 24 - 10000 active elements (0 deactivated)
21:56:50 INFO    opendrift.models.basemodel:1989: 2023-07-01 06:00:00 - step 13 of 24 - 10000 active elements (0 deactivated)
21:56:51 INFO    opendrift.models.basemodel:1989: 2023-07-01 06:30:00 - step 14 of 24 - 10000 active elements (0 deactivated)
22:00:39 INFO    opendrift.models.basemodel:1989: 2023-07-01 07:00:00 - step 15 of 24 - 10000 active elements (0 deactivated)
22:00:39 INFO    opendrift.models.basemodel:1989: 2023-07-01 07:30:00 - step 16 of 24 - 10000 active elements (0 deactivated)
22:00:39 INFO    opendrift.models.basemodel:1989: 2023-07-01 08:00:00 - step 17 of 24 - 10000 active elements (0 deactivated)
22:00:39 INFO    opendrift.models.basemodel:1989: 2023-07-01 08:30:00 - step 18 of 24 - 10000 active elements (0 deactivated)
22:00:39 INFO    opendrift.models.basemodel:1989: 2023-07-01 09:00:00 - step 19 of 24 - 10000 active elements (0 deactivated)
22:00:39 INFO    opendrift.models.basemodel:1989: 2023-07-01 09:30:00 - step 20 of 24 - 10000 active elements (0 deactivated)
22:00:45 INFO    opendrift.models.basemodel:1989: 2023-07-01 10:00:00 - step 21 of 24 - 10000 active elements (0 deactivated)
22:00:45 INFO    opendrift.models.basemodel:1989: 2023-07-01 10:30:00 - step 22 of 24 - 10000 active elements (0 deactivated)
22:00:46 INFO    opendrift.models.basemodel:1989: 2023-07-01 11:00:00 - step 23 of 24 - 10000 active elements (0 deactivated)
22:00:46 INFO    opendrift.models.basemodel:1989: 2023-07-01 11:30:00 - step 24 of 24 - 10000 active elements (0 deactivated)

o.plot(fast=True, markersize=.5)
![result](https://github.com/OpenDrift/opendrift/assets/149932569/69b54b4f-327d-4dc0-84af-851114d27afa)

I've been analyzing ocean currents using a ROMS server in South Korea and observed that despite the currents flowing north and south at approximately 0.5 m/s in a 6-hour cycle, I encountered the following results:

Even when changing the starting location or switching to the Leeway model, the simulation seemed to be influenced only by the wind direction.

For reference, I configured the HYCOM model using URLs like 'https://tds.hycom.org/thredds/dodsC/GLBy0.08/expt_93.0/FMRC/runs/GLBy0.08_930_FMRC_RUN_2023-11-17T12:00:00Z' for specific time periods and 'https://tds.hycom.org/thredds/dodsC/GLBy0.08/latest' for the latest version, but the results were the same.

I also checked on the HYCOM website to see if there might be no current model for that region, but I was able to confirm its existence through images.

How can I resolve this issue? I need some help.

**Comment by knutfrode** [comment]
Hi,
It could be that the HYCOM currents are not available from the thredds server at the time your were running the simulation.
If you change `loglevel` to `0` (debug), the log will tell you whether currents were loaded successfully or not.

Note: some models such as Leeway and OpenOil require that wind and currents are available, but OpenOil has 0 currents and 0 wind as default value in case it is not available, and it will thus not crash. But anyway it is alwas useful to check the log what is actually going on in your simulation.
